src/lib/device/vchunk.cpp

```diff
@@ -19,21 +19,21 @@
 namespace homestore {
     VChunk::VChunk(cshared< Chunk >& chunk) : internalChunk(chunk){}
 
-    void VChunk::set_user_private(const sisl::blob& data){
+    void VChunk::set_user_private(const sisl::blob& data) const{
```

Comment:
It is too bad to mark this function as const...

Reply:
a const function of a class means it will never modify the member of this class. the only member in VChunk is `cshared internalChunk` (`const shared_ptr`).

mark this function as const means we should not let `internalChunk` point to another chunk or be modified, that is it. but we could modify the `chunk` which it points to, which does not disobey the Semantics of const.

well, since we have already has `const shared_ptr` instead of `shared_ptr` for `internalChunk`, so actually it is unnecessary to add `const` to this function。is this your point here?

Reply:
it is very misleading that a "set" call could be "const",  which is weird.

Reply:
sure，let`s remove the const.